components/renku_data_services/session/core.py

```diff
@@ -31,14 +31,28 @@ def validate_unsaved_environment(
 def validate_environment_patch(patch: apispec.EnvironmentPatch) -> models.EnvironmentPatch:
     """Validate the update to a session environment."""
     data_dict = patch.model_dump(exclude_unset=True, mode="json")
+    working_directory = (
+        PurePosixPath(patch.working_directory)
+        if patch.working_directory
+        else RESET
+        if patch.working_directory == ""
+        else None
+    )
+    mount_directory = (
+        PurePosixPath(patch.mount_directory)
+        if patch.mount_directory
+        else RESET
+        if patch.mount_directory == ""
+        else None
+    )
```

Comment:
This and the above statement are a bit hard to parse. A match statement like this is roughtly the same number of lines but easier to understand:

```
    match patch.mount_directory:
        case "":
            mount_directory = RESET
        case str():
            mount_directory = PurePosixPath(patch.mount_directory)
        case _:
            mount_directory = None

```

Reply:
Done in 8914a70067a9b6c7332d276020c64226e17c0b48.